tests/tt_metal/tt_metal/api/CMakeLists.txt

```
@@ -34,27 +34,56 @@ set(UNIT_TESTS_API_SRC
     ${CMAKE_CURRENT_SOURCE_DIR}/test_worker_config_buffer.cpp
 )
 
-add_executable(unit_tests_api ${UNIT_TESTS_API_SRC})
-TT_ENABLE_UNITY_BUILD(unit_tests_api)
-
-target_link_libraries(
-    unit_tests_api
-    PRIVATE
-        test_metal_common_libs
-        Boost::smart_ptr
-)
-target_include_directories(
-    unit_tests_api
-    PRIVATE
-        "$<TARGET_PROPERTY:Metalium::Metal,INCLUDE_DIRECTORIES>"
-        ${PROJECT_SOURCE_DIR}/tests
-        ${PROJECT_SOURCE_DIR}/tests/tt_metal/tt_metal/common
-        ${CMAKE_CURRENT_SOURCE_DIR}
-        ${CMAKE_CURRENT_SOURCE_DIR}/common
-)
-set_target_properties(
-    unit_tests_api
-    PROPERTIES
-        RUNTIME_OUTPUT_DIRECTORY
-            ${PROJECT_BINARY_DIR}/test/tt_metal
+# Define the function to create test executables for each architecture
+function(create_unit_test_executable arch_name)
+    # Define the test executable name using the architecture name
+    set(exec_name unit_tests_api_${arch_name})
+    string(REPLACE "wormhole" "wormhole_b0" exec_name "${exec_name}")
+
+    # Create the test executable
+    add_executable(${exec_name} ${UNIT_TESTS_API_SRC})
+
+    # Enable unity build for the executable
+    TT_ENABLE_UNITY_BUILD(${exec_name})
+
+    # Link libraries
+    target_link_libraries(
+        ${exec_name}
+        PRIVATE
+            test_metal_common_libs
+            Boost::smart_ptr
+    )
+
+    # Set include directories
+    target_include_directories(
+        ${exec_name}
+        BEFORE
+        PRIVATE
+            ${PROJECT_SOURCE_DIR}/tt_metal/hw/inc/${arch_name}
+            "$<TARGET_PROPERTY:Metalium::Metal,INCLUDE_DIRECTORIES>"
+            ${PROJECT_SOURCE_DIR}/tests
+            ${PROJECT_SOURCE_DIR}/tests/tt_metal/tt_metal/common
+            ${CMAKE_CURRENT_SOURCE_DIR}
+            ${CMAKE_CURRENT_SOURCE_DIR}/common
+    )
+
+    # Set runtime output directory
+    set_target_properties(
+        ${exec_name}
+        PROPERTIES
+            RUNTIME_OUTPUT_DIRECTORY
+                ${PROJECT_BINARY_DIR}/test/tt_metal
+    )
+endfunction()
+
+# Define the architectures for which to create test executables
+set(ARCHITECTURES
+    "grayskull"
+    "wormhole"
+    "blackhole"
 )
+
+# Create a test executable for each architecture
+foreach(arch ${ARCHITECTURES})
```

(nit for legibility)

```suggestion
foreach(arch IN LISTS ARCHITECTURES)
```